spring-restdocs-core/src/main/java/org/springframework/restdocs/payload/JsonContentHandler.java

```diff
@@ -144,30 +147,4 @@ private boolean isEmpty(Object object) {
 		return ((List<?>) object).isEmpty();
 	}
 
-	@Override
-	public Object determineFieldType(FieldDescriptor fieldDescriptor) {
```

Comment:
I think this logic should remain here. It's subtle, but there's a difference between resolving the type of a field (which is what the field type resolver does) and determining the type of a field. It's only the latter that considers the type that's specified on the descriptor and checks whether or not they match. Ideally the former would only use the path (it currently looks at whether the field is optional too). I'd like to keep the current level of separation.